src/linse/typedsequence.py

```diff
@@ -91,15 +91,31 @@ def __init__(self, iterable, sep=" + "):
         self.sep = sep
 
     def __add__(self, other):
-        return Word(str(self) + self.sep + str(other))
+        if self and other:
+            self.morphemes.append(Morpheme(other))
+            return Word(str(self) + self.sep + str(other))
+        else:
+            return Word(str(self) or str(other))
+
+    def __iadd__(self, other):
+        return self.__add__(other)
```

Comment:
Subclassing mutable types is always a bit complicated - in particular if you have internal state, i.e. stuff you did in `__init__`. The alternative could be overwriting the data access methods, and doing the stuff you do in `__init__` now only when the data is accessed. This might be slower, though, and could lead to issues that are harder to debug.